### PR TITLE
Test/add missing testcase stakeibc

### PR DESCRIPTION
**Pull request by hoangdv2429**
Closes: #XXX

## Author's Checklist

I have...

- [ ] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

**Review by hieuvubk — COMMENTED** [review]
LGTM just need to remove some comment, @ThanhNhann what do u think?

File: x/stakeibc/keeper/msg_server_register_host_zone_test.go [hunk 1]
```diff
@@ -346,4 +346,4 @@ func (s *KeeperTestSuite) TestRegisterHostZone_CannotRegisterRedemptionAccount()
 	expectedErrMsg += "on connection connection-0 for owner GAIA.REDEMPTION: active channel already set for this owner: "
 	expectedErrMsg += "failed to register host zone"
 	s.Require().EqualError(err, expectedErrMsg, "can't register redemption account")
-}
+}
```
Comment (ThanhNhann) [inline x/stakeibc/keeper/msg_server_register_host_zone_test.go]: plz don't change this

File: x/stakeibc/keeper/msg_server_restore_interchain_account.go [hunk 2]
```diff
@@ -12,33 +12,29 @@ import (
...
 	owner := types.FormatICAAccountOwner(msg.ChainId, msg.AccountType)
 
 	// only allow restoring an account if it already exists
-	portID, err := icatypes.NewControllerPortID(owner)
-	if err != nil {
-		errMsg := fmt.Sprintf("could not create portID for ICA controller account address: %s", owner)
-		k.Logger(ctx).Error(errMsg)
-		return nil, err
-	}
+	portID, _ := icatypes.NewControllerPortID(owner)
```
Comment (ThanhNhann) [inline x/stakeibc/keeper/msg_server_restore_interchain_account.go:28]: this means err have checked in the `icatypes.NewControllerPortID()` function, right?

File: x/stakeibc/keeper/msg_server_restore_interchain_account.go [hunk 3]
```diff
@@ -12,33 +12,29 @@ import (
...
 	if !exists {
-		errMsg := fmt.Sprintf("ICA controller account address not found: %s", owner)
+		errMsg := fmt.Sprintf("ICA controller account address not found: %s", owner) //done
 		k.Logger(ctx).Error(errMsg)
 		return nil, sdkerrors.Wrapf(types.ErrInvalidInterchainAccountAddress, errMsg)
 	}
 
 	if err := k.ICAControllerKeeper.RegisterInterchainAccount(ctx, hostZone.ConnectionId, owner); err != nil {
-		k.Logger(ctx).Error(fmt.Sprintf("unable to register %s account : %s", msg.AccountType.String(), err))
+		k.Logger(ctx).Error(fmt.Sprintf("unable to register %s account : %s", msg.AccountType.String(), err)) //adding (already have, but still: missing 2 cases)
```
Comment (ThanhNhann) [inline x/stakeibc/keeper/msg_server_restore_interchain_account.go]: what is the mean of these comments?

File: x/stakeibc/keeper/msg_server_restore_interchain_account.go (hunk 2 above)
Comment (hoangdv2429) [reply x/stakeibc/keeper/msg_server_restore_interchain_account.go:28]: the hostzone check make sure this cannot provide any err. Also in the below function when running RegisterHostzone, this check will also happend again. So no need for this check.

File: x/stakeibc/keeper/msg_server_restore_interchain_account.go (hunk 3 above)
Comment (hoangdv2429) [reply x/stakeibc/keeper/msg_server_restore_interchain_account.go]: I should delete this after checking all conditions are fullfilled.

File: x/stakeibc/keeper/msg_server_register_host_zone_test.go (hunk 1 above)
Comment (hoangdv2429) [reply x/stakeibc/keeper/msg_server_register_host_zone_test.go]: yes

File: x/stakeibc/keeper/msg_server_restore_interchain_account.go (hunk 3 above)
Comment (ThanhNhann) [reply x/stakeibc/keeper/msg_server_restore_interchain_account.go]: u should delete this or anything else like this before you create a pr, it should be in your local not the remote man

Comment (hoangdv2429) [reply x/stakeibc/keeper/msg_server_restore_interchain_account.go]: Sorry, I created this PR only for quick a review. I am not intending to merge this., I will remove the comments and create a PR like this to your branch.

**Pull request closed without merging**